bundles/org.eclipse.core.databinding.property/src/org/eclipse/core/internal/databinding/property/value/MapDelegatingValueObservableMap.java

```diff
@@ -270,12 +268,11 @@ public Object getValueType() {
 
 	private Set<K> keysFor(I masterValue) {
 		Set<K> keys = new HashSet<>();
-
-		for (Map.Entry<K, I> entry : masterMap.entrySet()) {
+		masterMap.entrySet().forEach(entry -> {
```

Comment:
same here what is the benefit of this construct? looking at the code I would say a more modern approach would be  using stream with filter and collect them into a set.

Reply:
made those changes as per your suggestion. please have a look.